### PR TITLE
Daisy: major resource restructure.

### PR DESCRIPTION
**Pull request by crunk1**
- created a base resource struct
- Create{Disk|Image|Instances} are now slices of Disk|Image|Instance
- Population and Validation moved to the resource structs

### CODE REVIEW

File: daisy/disk.go [hunk 1]
```diff
@@ -15,39 +15,127 @@
...
+	SizeGb string `json:"sizeGb,omitempty"`
+}
+
+// MarshalJSON is a hacky workaround to prevent Disk from using compute.Disk's implementation.
+func (d *Disk) MarshalJSON() ([]byte, error) {
+	return json.Marshal(*d)
+}
+
+func (d *Disk) populate(ctx context.Context, s *Step) dErr {
+	errs := d.resource.populateWithZone(ctx, s, &d.Name, &d.Zone)
```
Comment (adjackura) [inline daisy/disk.go]: d.Name, d.Zone, errs = d.resource.populateWithZone(ctx, s, d.Name, d.Zone)

or don't return name at all and instead:
d.Name = d.resource.realName

File: daisy/resource.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+	r.daisyName = *name
+	*name = r.realName
+	r.Project = strOr(r.Project, s.w.Project)
+	return errs
+}
+
+func (r *resource) populateWithZone(ctx context.Context, s *Step, name *string, zone *string) dErr {
+	errs := r.populate(ctx, s, name)
+	*zone = strOr(*zone, s.w.Zone)
+	return errs
```
Comment (adjackura) [inline daisy/resource.go]: return name, zone, errs

File: daisy/resource.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+		r.realName = s.w.genName(*name)
+	}
+	r.daisyName = *name
+	*name = r.realName
+	r.Project = strOr(r.Project, s.w.Project)
+	return errs
+}
+
+func (r *resource) populateWithZone(ctx context.Context, s *Step, name *string, zone *string) dErr {
+	errs := r.populate(ctx, s, name)
```
Comment (adjackura) [inline daisy/resource.go]: name, errs := r.populate(ctx, s, name)

File: daisy/resource.go [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+		r.realName = *name
+	} else if r.RealName != "" {
+		r.realName = r.RealName
+	} else {
+		r.realName = s.w.genName(*name)
+	}
+	r.daisyName = *name
+	*name = r.realName
+	r.Project = strOr(r.Project, s.w.Project)
+	return errs
```
Comment (adjackura) [inline daisy/resource.go]: return name, errs

File: daisy/resource.go (hunk 2 above)
Comment (crunk1) [reply daisy/resource.go]: I merged populateWithZone into populate.

File: daisy/resource.go (hunk 3 above)
Comment (crunk1) [reply daisy/resource.go]: Done.

File: daisy/resource.go (hunk 4 above)
Comment (crunk1) [reply daisy/resource.go]: Done.

File: daisy/disk.go (hunk 1 above)
Comment (crunk1) [reply daisy/disk.go]: Went with the first.

**Pull request merged**